src/blocks/collapsible/index.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Server-side rendering of the `editorial/collapsible` block.
+ *
+ * @package BU_Blocks
+ */
+
+namespace BU\Plugins\BU_Blocks\Blocks\Collapsible;
+
+add_action( 'init', __NAMESPACE__ . '\\register_block' );
+
+/**
+ * Registers the `editorial/collapsible` block on the server.
+ */
+function register_block() {
+	register_block_type(
+		'bu/collapsible',
+		array(
+			'attributes'      => [
+				'title' => [
+					'type'     => 'string',
+				],
+				'level' => [
+					'type'     => 'number',
+					'default'  => 2,
+				],
+				'isOpen' => [
+					'type'     => 'boolean',
+					'default'  => false,
+				],
+				'iconStyle' => [
+					'type'     => 'string',
+					'default'  => 'plus-minus',
+				],
+				'customMarginBottom' => [
+					'type'     => 'boolean',
+					'default'  => false,
+				],
+				'marginBottom' => [
+					'type'     => 'number',
+					'default'  => 0,
+				],
+				'id' => [
+					'type'     => 'string',
+				],
+			],
+			'render_callback' => __NAMESPACE__ . '\\render_block',
+		)
+	);
+}
+
+/**
+ * Renders the `bu/collapsible` block.
+ *
+ * @param array  $attributes The block attributes.
+ * @param string $content    The block content.
+ * @return string Returns the b
+ */
+function render_block( $attributes, $content ) {
+	$attr = wp_parse_args(
+		$attributes,
+		[
+			'title'              => '',
+			'level'              => '2',
+			'isOpen'             => false,
+			'iconStyle'          => 'plus-minus',
+			'className'          => 'is-style-plain',
+			'customMarginBottom' => false,
+			'marginBottom'       => 0,
+			'id'                 => 'bu-collapsible-1',
+		]
+	);
+	$classes = [
+		'wp-block-bu-collapsible',
+		$attr['isOpen'] ? 'is-open' : 'is-closed',
+		"icon-style-{$attr['iconStyle']}",
+		$attr['className'],
+	];
+	$styles = [
+		'margin-bottom' => $attr['customMarginBottom'] ? "{$attr['marginBottom']}px" : '0',
+	];
+	$get_styles = function( $styles ) {
+		$out = '';
+		foreach ( $styles as $k => $v ) {
+			$out .= "{$k}:{$v}";
+		}
+		return $out;
+	};
+	ob_start();
+	?>
+
+	<div class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>" style="<?php echo esc_attr( $get_styles( $styles ) ); ?>" id="<?php echo esc_attr( $attr['id'] ); ?>">
+		<button class="bu-block-collapsible-toggle" id="<?php echo esc_attr( "{$attr['id']}-toggle"); ?>">
+			<<?php echo esc_html( "h{$attr['level']}" );?> class="bu-collapsible-heading"><?php echo wp_kses_post( $attr['title'] ); ?></<?php echo esc_html( "h{$attr['level']}" );?>>
+		</button>
```

Comment:
Can we have the heading wrap the button instead of the button wrapping the heading, going off the pattern in Inclusive Components? https://inclusive-components.design/collapsible-sections/ This struck me as a little odd.

That said, if there's a good accessibility reason behind this - please let me know so I can take note!

Reply:
That make sense.
I updated the toggle sturcture.
You can test/check it here: https://10up-dev1.cms-devl.bu.edu/com-staging/10up-collapsible-bu-blocks/

Let me know if there's other adjustment needed.
Thank you

Reply:
this change looks good to me.